test/features/steps/steps.py

```diff
@@ -0,0 +1,59 @@
+from behave import when, then
+from behaving.web.steps import *  # noqa: F401, F403
+from behaving.personas.steps import *  # noqa: F401, F403
+from behaving.web.steps.url import when_i_visit_url
+from behaving.mail.steps import *
+import random
+import email
+import quopri
+
+@when('I go to homepage')
+def go_to_home(context):
+    when_i_visit_url(context, '/')
+
+
+@when('I log in')
+def log_in(context):
+
+    assert context.persona
+    context.execute_steps(u"""
+        When I fill in "login" with "$name"
+        And I fill in "password" with "$password"
+        And I press the element with xpath "//button[contains(string(), 'Login')]"
+    """)
+
+@when('I log in and go to datarequest page')
+def log_in_go_to_datarequest(context):
+
+    assert context.persona
+    context.execute_steps(u"""
+        When I go to homepage
+        And I click the link with text that contains "Log in"
+        And I log in
+        And I go to datarequest page
+    """)
+
+@when('I go to datarequest page')
+def go_to_datarequest(context):
+    when_i_visit_url(context, '/datarequest')
+
+@when('I fill in title with random text')
+def title_random_text(context):
+
+    assert context.persona
+    context.execute_steps(u"""
+        When I fill in "title" with "Test Title {0}"
+    """.format(random.randrange(1000)) )
+
+# The default behaving step does not convert base64 emails
+# Modifed the default step to decode the payload from base64
+@then(u'I should receive an email at "{address}" containing "{text}"')
+def should_receive_email_containing_text(context, address, text):
+    def filter_contents(mail):
```

Comment:
maybe useful to not have inside this function, and wrapped in a if just in case the contents is of other states. (maybe use a python library for email reading? as there is at least 5 type of content encoding)

https://www.w3.org/Protocols/rfc1341/5_Content-Transfer-Encoding.html
Content-Transfer-Encoding := "BASE64" / "QUOTED-PRINTABLE" / "8BIT"   / "7BIT" / "BINARY" / x-token

content-transfer-encoding: quoted-printable 
looks like this
```
<!DOCTYPE html>=0D
<html>=0D
<head><meta name=3D"viewport" content=3D"width=3Ddevice-width"><meta http-e=
quiv=3D"Content-Type" content=3D"text/html; charset=3DUTF-8">=0D
etc
```